federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -406,34 +406,33 @@ func (s *ServiceController) deleteClusterService(clusterName string, cachedServi
 
 func (s *ServiceController) ensureClusterService(cachedService *cachedService, clusterName string, service *v1.Service, client *kubeclientset.Clientset) error {
 	var err error
-	var needUpdate bool
 	for i := 0; i < clientRetryCount; i++ {
 		svc, err := client.Core().Services(service.Namespace).Get(service.Name)
 		if err == nil {
 			// service exists
 			glog.V(5).Infof("Found service %s/%s from cluster %s", service.Namespace, service.Name, clusterName)
-			//reserve immutable fields
-			service.Spec.ClusterIP = svc.Spec.ClusterIP
+			// we tend to update the new spec and
+			// current service spec(might have random cluster ip and node port) in the cluster identical
+
+			// cluster ip is not specified, update the auto assigned one to new service spec
+			if service.Spec.ClusterIP == "" {
+				service.Spec.ClusterIP = svc.Spec.ClusterIP
```

Comment:
@mfanjie You're overwriting the ClusterIP of the federated service passed in by the caller here. I think you need to do this differently, e.g. only overwrite svc.Spec.ClusterIP if service.spec.ClusterIP is set.